velox/exec/TableWriter.cpp

```diff
@@ -101,34 +101,13 @@ RowVectorPtr TableWriter::getOutput() {
   }
   finished_ = true;
 
-  auto rowsWritten = std::dynamic_pointer_cast<FlatVector<int64_t>>(
-      BaseVector::create(BIGINT(), 1, pool()));
-  rowsWritten->set(0, numWrittenRows_);
-
-  std::vector<VectorPtr> columns = {rowsWritten};
-
-  // TODO Find a way to not have this Presto-specific logic in here.
-  if (outputType_->size() > 1) {
-    auto fragments = std::dynamic_pointer_cast<FlatVector<StringView>>(
-        BaseVector::create(VARBINARY(), 1, pool()));
-    fragments->setNull(0, true);
-    columns.emplace_back(fragments);
-
-    // clang-format off
-    auto commitContextJson = folly::toJson(
-        folly::dynamic::object
-            ("lifespan", "TaskWide")
-            ("taskId", driverCtx_->task->taskId())
-            ("pageSinkCommitStrategy", "NO_COMMIT")
-            ("lastPage", false));
-    // clang-format on
-
-    auto commitContext = std::make_shared<ConstantVector<StringView>>(
-        pool(), 1, false, VARBINARY(), StringView(commitContextJson));
-    columns.emplace_back(commitContext);
-  }
-
-  return std::make_shared<RowVector>(
-      pool(), outputType_, BufferPtr(nullptr), 1, columns);
+  TableWriterWriteInfo writeInfo(
```

Comment:
The names writeInfo and writerParameters are confusing me. WriterParameter sounds like input to a writer, but it is actual the result produced by a writer. Are these commitParameters? Would it make sense to call these commitInfo and commitParameters?

Reply:
Hi @mbasmanova , WriterParameters are input to a writer, https://github.com/facebookincubator/velox/pull/2586/files#diff-587fe1837b68791daefbadf6067b2362b2911480ccfe258615a78723dcf8d1c3R136-R147.

And at the same time, we need it in WriteProtocol::commit() and return it by TableWriter::getOutputs(). Being included in TableWriter::getOutputs() is unique behavior of Presto, so it is being used in presto_protocol/WriteProtocol.cpp, https://github.com/prestodb/presto/pull/18377/files#diff-d18ada7dc9e1ebc5521dd0f7e48e7df856c3474513a7320f3ee0680042ba3065R94-R96.

writerParameters is only part of the info we need to pass to commit() from writes, for Presto. So we have the abstract WriteInfo. And generally different system could have different implementation of WriteInfo to pass to commit().

Will it sound better if I call them CommitInfo and WriterParameters? WriterParameters is part of CommitInfo?

Reply:
@gggrace14 Ge, thank you for clarifying. CommitInfo sounds better.